formstack-baton-requests/src/main/scala/com/gu/identity/formstackbatonrequests/services/FormstackService.scala

```diff
@@ -97,26 +105,48 @@ object FormstackService extends FormstackRequestService with LazyLogging {
     case Left(e) => Left(e)
     case Right(submission) => Right(Some(submission))
   }
-  
+
+  def validateEmail(expectedEmail:String, submission: Submission) = submission.data.exists{
+    subData => extractEmails(subData.value).contains(expectedEmail)
+  }
+
   private def getSubmissions(
     submissionIdEmails: List[SubmissionIdEmail],
     accountToken: FormstackAccountToken,
-    encryptionPassword: String): Either[Throwable, List[Submission]] = {
-    val submissionResults: Either[Throwable, List[Option[Submission]]] = submissionIdEmails.traverse { submissionIdEmail =>
+    encryptionPassword: String): Either[Throwable, SubmissionsResponse] = {
+    val submissionResults: Either[Throwable, List[getSubmissionResult]] = submissionIdEmails.traverse { submissionIdEmail =>
+
       val response =
-        Http(s"https://www.formstack.com/api/v2/submission/${submissionIdEmail.submissionId}.json")
+        http(s"https://www.formstack.com/api/v2/submission/${submissionIdEmail.submissionId}.json")
           .header("Authorization", accountToken.secret)
           .param("encryption_password", encryptionPassword)
           .asString
           
       if(!response.is2xx) {
         logger.error(response.body)
       }
-      
-          decodeIfNotSkippableError[Submission](response)
+
+      decodeIfNotSkippableError[Submission](response).map{
+        case None => Skipped(submissionIdEmail)
+        case Some(submission) =>
+          //validate the submission we found in formstack actually contains references to the email we were looking for
```

Comment:
this is always useful but in particular now that we are going to be looking at submissions on different formstack accounts than the one they were originally discovered in we really want to make extra sure it's not a completely unrelated submission that happens to have the same id. This is probably unlikely to happen but it's good to not have to worry about it